src/sql/parts/notebook/notebookActions.ts

```diff
@@ -38,6 +40,42 @@ export class AddCellAction extends Action {
 	}
 }
 
+export class TrustedAction extends Action {
+	private static readonly trustLabel = localize('trustLabel', 'Trusted');
+	private static readonly notTrustLabel = localize('untrustLabel', 'Not Trusted');
+	private static readonly alreadyTrustedMsg = localize('alreadyTrustedMsg', 'Notebook is already trusted.');
+
+	constructor(
+		id: string, isTrusted: boolean
+	) {
+		if (isTrusted) {
+			super(id, TrustedAction.trustLabel, 'notebook-button icon-trusted');
+		}
+		else {
+			super(id, TrustedAction.notTrustLabel, 'notebook-button icon-notTrusted');
+		}
+	}
+	public run(context: NotebookComponent): TPromise<boolean> {
+		return new TPromise<boolean>((resolve, reject) => {
+			try {
+				if (context._model.trustedMode) {
+					 let notificationService = new NotificationService();
+					 notificationService.notify({severity: Severity.Info, message: TrustedAction.alreadyTrustedMsg});
+				}
+				else {
+					context.updateModelTrustDetails(!context._model.trustedMode);
+					this._setLabel(TrustedAction.trustLabel);
+					this._setClass('notebook-button icon-trusted');
```

Comment:
Consider making this class definition a constant since you reference in 2 places.

Reply:
Was working on this after pushing the PR. I probably needed some patience :).